libraries/AP_HAL_ChibiOS/hwdef/ARK_FPV/hwdef.dat

```diff
@@ -0,0 +1,215 @@
+# Processed by chibios_hwdef.py
+
+include common.inc
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 2
+
+# Bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# Use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+define HAL_STORAGE_SIZE 32768
+STORAGE_FLASH_PAGE 14
+
+# To be compatible with the px4 bootloader we need
+# to use a different RAM_MAP
+env USE_ALT_RAM_MAP 1
+
+SERIAL_ORDER OTG1 UART7 UART5 USART1 USART2 UART4 USART6 OTG2
+
+# Debug console
+STDOUT_SERIAL SD3
+STDOUT_BAUDRATE 57600
+
+# Default to all pins low to avoid ESD issues
+DEFAULTGPIO OUTPUT LOW PULLDOWN
+
+# Telem1 - Telem
+PE8  UART7_TX  UART7
+PF6  UART7_RX  UART7
+PF8  UART7_RTS UART7
+PE10 UART7_CTS UART7
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_MAVLink2
+define DEFAULT_SERIAL1_BAUD 115
+
+# Telem2 - VTX (DJI Air Unit)
+PC12 UART5_TX UART5
+PD2  UART5_RX UART5
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_MSP_DisplayPort
+define DEFAULT_SERIAL2_BAUD 115
+
+# GPS1
+PB6 USART1_TX USART1
+PB7 USART1_RX USART1
+define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_GPS
+
+# Telem3 - VTX (DJI Air Unit, RX only)
+PA3 USART2_RX USART2
+PD5 USART2_TX USART2
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_None # TODO: DJI HDL?
+
+# ESC telem RX only
+PH13 UART4_TX UART4 NODMA
+PH14 UART4_RX UART4
+define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_ESCTelemetry
+define DEFAULT_SERIAL5_BAUD 115
+define HAL_SERIAL_ESC_COMM_ENABLED 1
+define HAL_WITH_ESC_TELEM 1
+
+# RC Input
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_RCIN
+
+# TODO: build was failing
+# define DEFAULT_SERIAL7_PROTOCOL SerialProtocol_SLCAN
+
+# ADCs
+PA0 SCALED1_V3V3 ADC1 SCALE(2)
+PB1 VDD_5V_SENS ADC1 SCALE(2)
+# TODO: 12V monitor
+# PA4 SCALED2_V3V3 ADC1 SCALE(7.66)
```

Comment:
Do non int values work here? Is this the right way to do it?
![Screenshot from 2024-12-11 17-26-15](https://github.com/user-attachments/assets/34532131-311f-4d3a-a4da-b397c87e42a3)

Reply:
yes this works